docs/enhancements/install-strategy.md

```diff
@@ -0,0 +1,233 @@
+
+
+## Motivation
+
+Hive is presently struggling with how to design the API to meet the needs of provisioning standalone, agent based, and hypershift clusters. The current code base evolved quite specific to running openshift-install pods and has some growing pains to accomodate new paths to provision clusters. Changes for one raise a lot of questions and inconsistencies for the others as everything is lumped into ClusterDeployment. The line between day 1 and day 2 is very unclear and mostly just enforced via increasingly complicated validating webhooks.
+
+### Goals
+
+- Clarify Hive APIs. (day 1 vs day 2)
+- Breakout clusterDeployment.spec.installStrategy.agent to a separate AgentClusterInstall CRD. This is still guarded by an alpha feature gate and thus we should still be able to move it around.
+- Provide space for implementations to define the spec and status they need without causing uncomfortable complications for other implementations.
+- Do not break API compatability for existing standalone OpenShift cluster provisioning. We hope to slowly migrate those installs to this system by offering it in parallel and eventually deprecating the old, but for now the focus is on breaking out the AgentClusterInstall CRD.
+- Plan for a future of integration of ClusterInstalls with ClusterPools. (see Future Considerations)
+
+### Non-Goals
+
+- Replace ClusterProvision. This CRD represents an attempt to install a standalone cluster, it's highly specific to running openshift-install in a pod, and we do not want to modify anything in this area. Even if we move standalone provisions to an OpenShiftClusterInstall, ClusterProvision will likely be used beneath it as the retry mechanism.
+- Implementing retries within Hive itself. Retries will be left to specific implementations of ClusterInstall interface, if desired.
+- Implementing ClusterPool support for ClusterInstalls. (see Future Considerations)
+
+## Proposal
+
+Introduce a new ClusterInstall CRD interface/contract in Hive API, breaking out of the current ClusterDeployment.Spec and instead linking to one and use duck typing to interact. Initial implementation would be AgentClusterInstall. In the future we would hopefully migrate current openshift-install pods to OpenShiftClusterInstall, and add HypershiftClusterInstall, etc.
+
+Hive ClusterInstall implementations are a day 1 provisioning API. They would adhere to a contract with hive for what fields/conditions they must contain and update as the installation proceeds. Hive would copy relevant data back to the ClusterDeployment as appropriate at various stages of the provisioning.
+
+ClusterDeployment becomes a central index of clusters and kubeconfigs, used for day 2 management.
+
+AgentClusterInstall becomes a day 1 resource and will offer more flexibility for the agent controllers to define settings they need in isolation, and track their relevant status.
+
+
+### User Stories
+
+#### Story 1
+
+#### Story 2
+
+## Design Details
```

Comment:
we also need to include details on how these implementations providers can give hive access to these new cluster resources. i think we can use https://kubernetes.io/docs/reference/access-authn-authz/rbac/#aggregated-clusterroles to create an aggregate role for hive controllers.

Reply:
^^ @dgoodwin i think we need to include something to solve the permission issue.

Reply:
Added a section.